userspace/libscap/scap_savefile.c

```diff
@@ -137,25 +137,45 @@ int32_t scap_write_proc_fds(scap_t *handle, struct scap_threadinfo *tinfo, scap_
 	block_header bh;
 	uint32_t bt;
 	uint32_t totlen = MEMBER_SIZE(scap_threadinfo, tid);  // This includes the tid
+	uint32_t nfds = 0;
+	uint32_t idx = 0;
 	struct scap_fdinfo *fdi;
 	struct scap_fdinfo *tfdi;
 
 	//
-	// First pass of the table to calculate the length
+	// First pass of the table to calculate the size
+	//
+	HASH_ITER(hh, tinfo->fdlist, fdi, tfdi)
+	{
+		nfds++;
+	}
+
+	uint32_t* lengths = calloc(nfds, sizeof(uint32_t));

```

Comment:
There seem to be several occurrences where the memory is not freed upon error. I see the same behavior in other parts of the PR where calloc/malloc is used, so I'm going to report this feedback just once here.

Reply:
You're definitely right, thanks for catching those.